pkg/component/component_test.go

```diff
@@ -29,6 +31,281 @@ import (
+
+// 			}
+
+// 			got := getComponentFrom(mockLocalConfigProvider, tt.componentType)
+// 			if !reflect.DeepEqual(got, tt.want) {
+// 				t.Errorf("getComponentFrom() = %v, want %v", got, tt.want)
+// 			}
+
+// 		})
+// 	}
+// }
+
+func TestTestifyGetComponentFrom(t *testing.T) {
+	debport := 2439
+	tests := []struct {
+		name          string
+		isEnvInfo     bool
+		componentName string
+		componentType string
+		env           []envinfo.EnvInfoURL
+		cmpSetting    envinfo.ComponentSettings
+		defaultPort   int
+		want          Component
+	}{
+		{
+			name:          "Case 1: Get component when env info file exists with nil debug port",
+			isEnvInfo:     true,
+			componentType: "nodejs",
+			env: []envinfo.EnvInfoURL{
+				{
+					Name: "abcde",
+					Port: 5858,
+				},
+			},
+			cmpSetting: envinfo.ComponentSettings{
+				Name:      "frontend",
+				Project:   "test",
+				URL:       &[]envinfo.EnvInfoURL{},
+				AppName:   "testing",
+				DebugPort: nil,
+			},
+			defaultPort: 5859,
+			want: Component{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Component",
+					APIVersion: "odo.dev/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "frontend",
+				},
+				Spec: ComponentSpec{
+					Type: "nodejs",
+				},
+				Status: ComponentStatus{},
+			},
+		},
+
+		{
+			name:          "Case 2: Get component when env info file exists with debug port and url value",
+			isEnvInfo:     true,
+			componentType: "nodejs",
+			env: []envinfo.EnvInfoURL{
+				{
+					Name: "abcde",
+					Port: 5858,
+				},
+			},
+			cmpSetting: envinfo.ComponentSettings{
+				Name:    "frontend",
+				Project: "test",
+				URL: &[]envinfo.EnvInfoURL{
+					{
+						Name: "abc",
+						Port: 5856,
+					},
+				},
+				AppName:   "testing",
+				DebugPort: &debport,
+			},
+			defaultPort: 5859,
+			want: Component{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Component",
+					APIVersion: "odo.dev/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "frontend",
+				},
+				Spec: ComponentSpec{
+					Type: "nodejs",
+				},
+				Status: ComponentStatus{},
+			},
+		},
+
+		{
+			name:          "Case 3: Get component when env info file does not exists",
+			isEnvInfo:     false,
+			componentType: "nodejs",
+			want:          Component{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			envInfoSpc := envinfo.EnvSpecificInfo{
+				EnvinfoFileExists: tt.isEnvInfo,
+			}
+
+			mockLocalConfig := &mocks.LocalConfigProvider{}
+
+			envFileExists := (reflect.ValueOf((mockLocalConfig.On("Exists").Return(envInfoSpc.EnvinfoFileExists)).ReturnArguments[0])).Interface().(bool)
+
+			if envFileExists {
+				var debugPort int
+				var urls []envinfo.EnvInfoURL
+
+				envInfo := envinfo.EnvInfo{
+					ComponentSettings: tt.cmpSetting,
+				}
+
+				cmpN := (reflect.ValueOf((mockLocalConfig.On("GetName").Return(envInfo.ComponentSettings.Name)).ReturnArguments[0])).Interface().(string)
```

Comment:
Will do cleanup and modify the variable name after the final decision on the tests. This is just a kind of example, how we should write the unit test and create the mocking